nx_parallel/tests/test_get_chunks.py

```diff
@@ -0,0 +1,84 @@
+# smoke tests for all functions supporting `get_chunks` kwarg
+
+import inspect
+import importlib
+import networkx as nx
+import nx_parallel as nxp
+import random
+import types
+import math
+
+
+def get_all_functions(package_name="nx_parallel"):
+    """Returns a dictionary where the keys are the function names in a given Python package, and the values are dictionaries containing the function's keyword arguments and positional arguments."""
+    package = importlib.import_module(package_name)
+    functions = {}
+
+    for name, obj in inspect.getmembers(package, inspect.isfunction):
+        if not name.startswith("_"):
+            args, kwargs = inspect.getfullargspec(obj)[:2]
+            functions[name] = {"args": args, "kwargs": kwargs}
+
+    return functions
+
+
+def get_functions_with_get_chunks():
+    """Returns a list of functions with the `get_chunks` kwarg."""
+    all_funcs = get_all_functions()
+    get_chunks_funcs = []
+    for func in all_funcs:
+        if "get_chunks" in all_funcs[func]["args"]:
+            get_chunks_funcs.append(func)
+    return get_chunks_funcs
+
+
+def test_get_chunks():
+    def random_chunking(nodes):
+        _nodes = list(nodes).copy()
+        random.seed(42)
+        random.shuffle(_nodes)
+        num_chunks = nxp.cpu_count()
+        num_in_chunk = max(len(_nodes) // num_chunks, 1)
+        return nxp.chunks(_nodes, num_in_chunk)
+
+    get_chunks_funcs = get_functions_with_get_chunks()
+    ignore_funcs = [
```

Comment:
the functions that require some argument(s) other than `G` to be passed in the function call are ignored. 

Should the `get_chunk` for these be tested in `test_get_chunks.py` or in a separate `test` folder in the respective modules to which these functions belong?

I think it would be better to have all the `get_chunk` smoke tests in one place. But, I don't know how many functions in networkx would fall in this category(i.e. algos that require args other than just `G`). LMK what you think.

Reply:
What do you think about having tests of the functionality of get_chunks as a function? Those would be separate from any tests that the other functions are using get_chunks correctly (which is I think what the smoke tests are doing... right?).
 
From a theoretical perspective, if the smoke tests are testing that the nxp functions are using get_chunks correctly, that should probably be in a test module for that function.
And how many functions in networkx have parameters other than G is less important than how many functions supported by nx-parallel do. I suspect that most functions do have some input other than G. 
 
Do we need to test all the functions' get_chunk parameter? I guess it is a check of whether each function is using chunks correctly.
 
Manually listing functions that can (or can't) be run with G as the only parameter doesn't scale nicely to all networkx functions. Another option is to set up test files for each module -- similar to networkx. But that adds a lot of testing files. Yet another option is to find some way to run the networkx test suite with all calls to nx-parallel setting a specified value for the get_chunks parameter. That might not be practical, and I haven't thought much about it. This is tricky... :)

Reply:
IMO it would be ok to have all the smoke tests for all the functions' `get_chunks` in one file, right now, bcoz we don't have a lot of functions in `ignore_func`, and having so many test files with just one smoke test(all of which can be summed up in a loop like this) seems unnecessary and redundant. But this is definitely not scalable, so, we can handle this differently once we have a bunch of functions in nx-parallel that require args other than just `G`. And we can have separate `tests` folders if necessary, like we currently have for `betweenness_centrality`(which not just tests `get_chunks` but also shows how custom chunking can be useful). LMK what you think of this.

And should we add the smoke tests for the functions in `ignore_func` in this file only or in separate `tests` folders in respective modules?

> What do you think about having tests of the functionality of get_chunks as a function? Those would be separate from any tests that the other functions are using get_chunks correctly (which is I think what the smoke tests are doing... right?).

I think we can test `get_chunks` separately for all functions in separate tests folders, once we have other additional tests for nx-parallel functions. But right now `get_chunks` smoke test is the only additional test for all nx-parallel functions.

> Yet another option is to find some way to run the networkx test suite with all calls to nx-parallel setting a specified value for the get_chunks parameter.

But, then this wouldn't check our default chunking.